### ISSUE TITLE
QgsVectorFileWriter.create() doesn't flush the output after deleting the reference

### ISSUE THREAD

**Issue by kadirsahbaz** [qgis/QGIS#35021]
When I use the following script, the output is not flushed after deleting the reference (`writer`). It's flushed only after exiting QGIS.

```
file_path = "c:/users/user/desktop/aa.shp"
crs = QgsCoordinateReferenceSystem("EPSG:4326") # WGS84
context = QgsCoordinateTransformContext()
options = QgsVectorFileWriter.SaveVectorOptions()
options.driverName = "ESRI Shapefile"

fields = QgsFields()
field = QgsField( 'newColumn', QVariant.Double )
fields.append(field)

writer = QgsVectorFileWriter.create(file_path,  fields, QgsWkbTypes.Point, crs, context, options)

f = QgsFeature(fields)
f["newColumn"] = 1.1
f.setGeometry(QgsGeometry.fromPointXY(QgsPointXY(0, 0)))
writer.addFeature(f)

del writer
```

QGIS version: 3.12.0-București
QGIS code revision: cd141490ec
Compiled against Qt: 5.11.2
Running against Qt: 5.11.2
Compiled against GDAL/OGR: 3.0.4
Running against GDAL/OGR: 3.0.4
Compiled against GEOS: 3.8.0-CAPI-1.13.1
Running against GEOS: 3.8.0-CAPI-1.13.1 
Compiled against SQLite: 3.29.0
Running against SQLite: 3.29.0
PostgreSQL Client Version: 11.5
SpatiaLite Version: 4.3.0
QWT Version: 6.1.3
QScintilla2 Version: 2.10.8
Compiled against PROJ: 6.3.1
Running against PROJ: Rel. 6.3.1, February 10th, 2020
OS Version: Windows 10 (10.0)
Active python plugins: db_manager; MetaSearch

**Comment by gioman** [comment]
@kadirsahbaz question or bug report?

**Comment by kadirsahbaz** [comment]
I'm not sure. I thought it was a bug. Since I was not sure, I added as a blank issue. 

There is a [post](https://gis.stackexchange.com/questions/353687/pyqgis-qgsvectorfilewriter-create-doesnt-flush-even-after-deleted) in GIS SE about that issue.

There is no test about `create` method in source code (python side). I've examined source code (C++). But because I'm not a C++ geek,  I couldn't find anything eventually.
